cranelift/filetests/src/test_run.rs

```diff
@@ -24,6 +25,73 @@ pub fn subtest(parsed: &TestCommand) -> anyhow::Result<Box<dyn SubTest>> {
     Ok(Box::new(TestRun))
 }
 
+/// Builds a [TargetIsa] for the current host.
+///
+/// ISA Flags can be overridden by passing [Value]'s via `isa_flags`.
+fn build_host_isa(
+    infer_native_flags: bool,
+    flags: settings::Flags,
+    isa_flags: Vec<settings::Value>,
+) -> Box<dyn TargetIsa> {
+    let mut builder = cranelift_native::builder_with_options(infer_native_flags)
+        .expect("Unable to build a TargetIsa for the current host");
+
+    for value in isa_flags {
+        match value.kind() {
+            SettingKind::Bool => builder
+                .set(value.name, &format!("{}", value.as_bool().unwrap()))
+                .unwrap(),
+            _ => unimplemented!(),
+        }
+    }
+
+    builder.finish(flags).unwrap()
+}
+
+/// Checks if the host's ISA is compatible with the one requested by the test.
+fn is_isa_compatible(
+    context: &Context,
+    host: &dyn TargetIsa,
+    requested: &dyn TargetIsa,
+) -> Result<(), String> {
+    // If this test requests to run on a completely different
+    // architecture than the host platform then we skip it entirely,
+    // since we won't be able to natively execute machine code.
+    let host_arch = host.triple().architecture;
+    let requested_arch = requested.triple().architecture;
+    if host_arch != requested_arch {
+        return Err(format!(
+            "skipped {}: host can't run {:?} programs",
+            context.file_path, requested_arch
+        ));
+    }
+
+    // We need to check that the requested ISA does not have any flags that
+    // we can't natively support on the host.
+    let requested_flags = requested.isa_flags();
+    let bool_isa_flags = requested_flags
+        .iter()
+        .filter(|v| v.kind() == SettingKind::Bool);
+    for req_value in bool_isa_flags {
+        let requested = req_value.as_bool().unwrap_or(false);
+        let available_in_host = host
+            .isa_flags()
+            .iter()
+            .find(|val| val.name == req_value.name)
+            .and_then(|val| val.as_bool())
+            .unwrap_or(false);
+
+        if requested && !available_in_host {
+            return Err(format!(
+                "skipped {}: host does not support ISA flag {}",
+                context.file_path, req_value.name
+            ));
+        }
+    }
```

Comment:
It seems surprising to me to silently ignore non-bool flags here. How do you feel about this?

```suggestion
    for req_value in requested_flags {
        if let Some(requested) = req_value.as_bool() {
            let available_in_host = host
                .isa_flags()
                .iter()
                .find(|val| val.name == req_value.name)
                .and_then(|val| val.as_bool())
                .unwrap_or(false);

            if requested && !available_in_host {
                return Err(format!(
                    "skipped {}: host does not support ISA flag {}",
                    context.file_path, req_value.name
                )); 
            }
        } else {
            unimplemented!("ISA flag {} of kind {:?}", req_value.name, req_value.kind());
        }
    }
```